v2/azure-shared-resource.go

```diff
@@ -66,6 +71,11 @@ func NewAzureSharedResource(accountName, containerName string, sharedCapacity ui
 
 // This allows you to provide mocked objects for container and blob for unit tests.
 func (r *azureSharedResource) WithMocks(container AzureContainer, blob AzureBlob) AzureSharedResource {
```

Comment:
Not really, but it came after a lot of discussion with Megan, Nneoma, and others. We didn't really come up with a better way to do it.

Reply:
I brought the topic up again in the docs PR, because to me reading the testing doc makes it more clear why it feels wrong this way. The convo should probably be converged to this thread though and not scattered across two PRs, so I'll move that comment to here...

Reply:
From https://github.com/plasne/go-batcher/pull/22#discussion_r614832334 

> I mentioned this in another PR, but I have reservations about this mechanism for injecting mocks. I've not seen it done this way in any other codebase, and it strikes me as odd that using a testing backdoor like this would be a preferred approach.
> 
> Why not just use more traditional dependency injection techniques? Constructor injection, Property/Member injection, Factory/Provider, even just a simple application of Strategy pattern all seem like they would work here?
> 
> I'm hoping to learn some new aspect of this approach that makes it appealing, perhaps it's more idiomatic for Go or those other approaches have downsides in Go?

Reply:
We are going to do a separate PR on this.